### PR TITLE
Localise data views

### PR DESCRIPTION
**Pull request by emilyb7**
Makes sure that the views receive the correct translated content from the models

#102 

Also 
- reimplements page titles and some other small errors - #161 
- adds in some missing translations #163

### CODE REVIEW

**Comment by emilyb7** [comment]
waiting for #162 to be merged first

**Comment by emilyb7** [comment]
waiting for #147

**Comment by emilyb7** [comment]
@njsfield ready for your review now!

File: routes/middleware.js [hunk 1]
```diff
@@ -8,19 +8,20 @@ exports.initLocals = (req, res, next) => {
 
   const locals = res.locals;
   locals.user = req.user;
+
+  // Add your own local variables here
```
Comment (njsfield) [inline routes/middleware.js]: Remove redundant comment

File: routes/middleware.js [hunk 2]
```diff
@@ -8,19 +8,20 @@ exports.initLocals = (req, res, next) => {
...
   const viewColors = require('./setup/view-colors.js');
+  const view = req.url.split('/')[2];
```
Comment (njsfield) [inline routes/middleware.js]: Move back to below line 17 to match comment above it

File: routes/views/director_profile.js [hunk 3]
```diff
@@ -12,22 +12,27 @@ exports = module.exports = (req, res) => {
...
+        if (err || !director) {
+          res.locals.title = '404 error | YiMovi';
+          res.status(404).render('errors/404');
+          return;
+        }
+
+        locals.director = require('../helpers/localize_results.js')
+        .localizeResults(locals.locale, director);
+
+        locals.title = `${director.name} ${director.name_chn || ''} | YiMovi director profile`;
```
Comment (njsfield) [inline routes/views/director_profile.js]: This title key should be removed, the view builds the title name + static fields from JSON files, see [here](https://github.com/InFact-coop/YiMovi/blob/master/templates/views/director_profile.pug#L24)

File: routes/views/genre_profile.js [hunk 4]
```diff
@@ -20,7 +20,9 @@ exports = module.exports = (req, res) => {
         return;
       }
 
-      locals.genre = genre;
+      locals.genre = require('../helpers/localize_results.js')
+        .localizeResults(locals.locale, genre);
+
       locals.title = `${genre.name} films | YiMovi`;
```
Comment (njsfield) [inline routes/views/genre_profile.js]: Same as comment above, title should be removed.

File: routes/views/index.js [hunk 5]
```diff
@@ -14,13 +14,17 @@ exports = module.exports = (req, res) => {
       req.flash('success', res.locals.__.app.flash_messages.mail_sent);
     }
 
-    Theme.model.find().limit(8).sort('sortOrder').exec((err, themes) => {
+    Theme.model.find()
+      .where('name' + locals.locale === 'en' ? '' : `___${locals.locale}`).ne(null)
```
Comment (njsfield) [inline routes/views/index.js]: You should import the defaultLocale variable from the locales.js file

**Review by njsfield — CHANGES_REQUESTED** [review]
Nice work just some small changes, assign me again when ready

**Review by njsfield — APPROVED** [review]

**Review by njsfield — COMMENTED** [review]
(Accidentally set to approved before)

I'm changing to a general feedback review as I'd like to contribute to this project again later for contingency/next sprints. Assigning to Eoin for review

**Comment by njsfield** [comment]
Sorry I've assigned to you @emilyb7 first to review my comments, assign @des-des after.

**Comment by njsfield** [comment]
See #163

File: routes/views/director_profile.js [hunk 6]
```diff
@@ -8,26 +8,30 @@ exports = module.exports = (req, res) => {
...
+          res.status(404).render('errors/404');
+          return;
+        }
 
-      if (err || !director) {
-        res.locals.title = '404 error | YiMovi';
-        res.status(404).render('errors/404');
-        return;
-      }
+        locals.director = require('../helpers/localize_results.js')
```
Comment (des-des) [inline routes/views/director_profile.js]: Move require to top of file (consistent style)

File: routes/views/genre_profile.js [hunk 7]
```diff
@@ -10,26 +10,28 @@ exports = module.exports = (req, res) => {
...
-        return;
-      }
+        if (err || !genre) {
+          res.status(404).render('errors/404');
+          return;
+        }
 
-      locals.genre = genre;
-      locals.title = `${genre.name} films | YiMovi`;
+        locals.genre = require('../helpers/localize_results.js')
```
Comment (des-des) [inline routes/views/genre_profile.js]: move require to top

File: routes/views/index.js [hunk 8]
```diff
@@ -14,13 +16,19 @@ exports = module.exports = (req, res) => {
...
+    Theme.model.find()
+      // filter out untranslated results
+      .where('name' + (locals.locale === defaultLocale ? '' : `___${locals.locale}`)).ne(null)
+      .limit(8).sort('sortOrder')
+      .exec((err, themes) => {
+
+        if (err) return next(err);
 
-      if (err) return next(err);
+        locals.themes = require('../helpers/localize_results.js')
```
Comment (des-des) [inline routes/views/index.js]: here also the require

File: routes/views/theme_profile.js [hunk 9]
```diff
@@ -11,24 +11,26 @@ exports = module.exports = (req, res) => {
...
-          res.locals.title = '404 error | YiMovi';
-          res.status(404).render('errors/404');
-          return;
-        }
+        res.status(404).render('errors/404');
+        return;
       }
 
-      locals.theme = theme;
+      locals.theme = require('../helpers/localize_results.js')
```
Comment (des-des) [inline routes/views/theme_profile.js]: here

**Review by des-des — CHANGES_REQUESTED** [review]

**Review by des-des — APPROVED** [review]

**Pull request merged**